flowmachine/flowmachine/core/sqlalchemy_utils.py

```diff
@@ -0,0 +1,133 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import pandas as pd
+from sqlalchemy import Table, MetaData
+from sqlalchemy.dialects import postgresql
+from sqlalchemy.sql import Selectable
+
+
+def get_sqlalchemy_table_definition(fully_qualified_table_name, *, engine):
+    """
+    Return sqlalchemy Table object for table with the given name.
+
+    Parameters
+    ----------
+    fully_qualified_table_name : str
+        Fully qualified table name. Example: "events.calls"
+    engine : sqlalchemy.engine.Engine
+        SQLAlchemy engine to use for reading the table information.
+
+    Returns
+    -------
+    sqlalchemy.Table
+    """
+    metadata = MetaData()
+    if fully_qualified_table_name == "events.calls":
+        schema = "events"
+        table_name = "calls"
+    elif fully_qualified_table_name == "events.sms":
+        schema = "events"
+        table_name = "sms"
+    elif fully_qualified_table_name == "events.mds":
+        schema = "events"
+        table_name = "mds"
+    elif fully_qualified_table_name == "events.topups":
+        schema = "events"
+        table_name = "topups"
+    else:
+        raise NotImplementedError(
+            f"No sqlalchemy definition found for table: '{fully_qualified_table_name}'"
+        )
+
+    return Table(
+        table_name, metadata, schema=schema, autoload=True, autoload_with=engine
+    )
+
+
+def get_sql_string(query):
+    """
+    Return SQL string compiled from the given sqlalchemy query (using the PostgreSQL dialect).
+
+    Parameters
+    ----------
+    query : sqlalchemy.sql.Selectable
+        SQLAlchemy query
+
+    Returns
+    -------
+    str
+        SQL string compiled from the sqlalchemy query.
+    """
+    assert isinstance(query, Selectable)
+    compiled_query = query.compile(
+        dialect=postgresql.dialect(), compile_kwargs={"literal_binds": True}
+    )
+    sql = str(compiled_query)
+    return sql
+
+
+def get_query_result_as_dataframe(query, *, engine):
+    """
+    Run the given sqlalchemy query and return the result as a pandas DataFrame.
+
+    Parameters
+    ----------
+    query : sqlalchemy.sql.Selectable
+        The SQLAlchemy query to run.
+    engine : sqlalchemy.engine.Engine
+        SQLAlchemy engine to use for reading the table information.
+
+    Returns
+    -------
+    pandas.DataFrame
+        Data frame containing the result.
+    """
+    assert isinstance(query, Selectable)
+
+    with engine.connect() as con:
+        result = con.execute(query)
+
+    columns = [c.name for c in query.columns]
+    df = pd.DataFrame(result.fetchall(), columns=columns)
+    return df
+
+
+def get_sqlalchemy_column(table, column_str):
```

Comment:
Why does this not just have two arguments? column and label?

Reply:
Mainly because of the way it's being used in `EventTableSubset`, where the `column_str` comes from flowmachine. Because `column_str` can be either of the form `<column_name>` or `<column_name> AS <alias>` this string needs to be potentially broken apart, and the main point of this helper function is to do this here rather than outside. So it sounds like I need to find a better name for this function, to make it clearer. Any suggestions?

Reply:
I have renamed this function to make it a bit clearer that it has a pretty narrow purpose at the moment. I expect this to go away in the medium run once the sqlalchemy table and column definitions are created a bit more centrally (and simply passed into the classes that need them).